src/analysis/sensitivity.jl

```diff
@@ -406,11 +406,30 @@ function outcome_map(
     X::DataFrame,
     y::AbstractVecOrMat{<:Real},
     rule::Union{Function,BitVector,Vector{Int64}},
-    target_factors::Vector{String};
+    target_factors::Vector{String},
+    model_spec::DataFrame;
     S::Int64=20,
     n_boot::Int64=100,
     conf::Float64=0.95
 )::NamedDimsArray
+    if !all([tf in model_spec.fieldname for tf in target_factors])
+        missing_factor = .!([tf in model_spec.fieldname for tf in target_factors])
```

Comment:
I think they're different sizes because `missing_factor` is the inverse of what is inside `target_factors` (and so will have the length of target_factors) and `factors_to_assess` is what is inside `model_spec.fieldname` from `target_factors` (and so will have the same size as `model_spec.fieldname`, unless I've misinterpreted? But on checking they seem to be different sizes

Reply:
Oh I see, the check is to ensure the factors named in `target_factors` are valid or not.

I feel there's a simpler way of doing this but fine to ignore my comment then.